Tasks/JenkinsQueueJobV2/jobsearch.ts

```diff
@@ -43,23 +43,23 @@ export class JobSearch {
         if (!thisSearch.Initialized) { //only initialize once
             const apiTaskUrl: string = util.addUrlSegment(thisSearch.taskUrl, '/api/json?tree=downstreamProjects[name,url,color],lastBuild[number]');
             tl.debug('getting job task URL:' + apiTaskUrl);
-            request.get({ url: apiTaskUrl, strictSSL: thisSearch.queue.TaskOptions.strictSSL }, function requestCallBack(err, httpResponse, body) {
+
+            var request = new WebRequest();
+            request.uri = apiTaskUrl;
+            request.method = "GET";
+            request.headers = {
+                "Content-Type": "application/json; charset=utf-8",
+                "Authorization": 'Basic ' + new Buffer(thisSearch.queue.TaskOptions.username + ':' + thisSearch.queue.TaskOptions.password).toString('base64')
+            };
+        
+            var response = sendRequest(request).then(async (response: WebResponse) => {
```

Comment:
It doesn't appear that the `defer` is necessary anymore.  I think you can just:
```
return sendRequest...
```
and return `Promise.resolve()` or `Promise.reject()` where the defers are.  (It's easier for me to read than `defer`; others my feel differently, I'll leave it up to you.)

Reply:
I think we should actually move everything to async-await.  Happy to help with this if you want.

Reply:
Even better!

Reply:
Well, Kellie and I tried to pair program this for a while.  There's a lot of code that needs to change and we saw a lot of issues with the current handling of rejections and the like.  So I'm convinced we should tackle this separately.